core/chains/evm/forwarders/forwarder_manager.go

```diff
@@ -0,0 +1,299 @@
+			continue
+		}
+		f.setCachedSenders(fwdr.Address, senders)
+
+	}
+}
+func (f *FwdMgr) subscribeForwardersLogs(fwdrs []Forwarder) {
+	for _, fwdr := range fwdrs {
+		f.subscribeSendersChangedLogs(fwdr.Address)
+	}
+}
+
+func (f *FwdMgr) subscribeSendersChangedLogs(addr common.Address) {
+	f.logpoller.MergeFilter(
+		[]common.Hash{authorized_receiver.AuthorizedReceiverAuthorizedSendersChanged{}.Topic()},
+		addr)
+}
+
+func (f *FwdMgr) setCachedSenders(addr common.Address, senders []common.Address) {
+	f.cacheMu.Lock()
+	defer f.cacheMu.Unlock()
+	f.sendersCache[addr] = senders
+}
+
+func (f *FwdMgr) getCachedSenders(addr common.Address) ([]common.Address, bool) {
+	f.cacheMu.RLock()
+	defer f.cacheMu.RUnlock()
+	addrs, ok := f.sendersCache[addr]
+	return addrs, ok
+}
+
+func (f *FwdMgr) runLoop() {
+	tick := time.After(0)
+
+	for {
+		select {
+		case <-tick:
+			tick = time.After(utils.WithJitter(AuthLogCheckInterval))
+			addrs := f.collectAddresses()
+			logs, err := f.logpoller.LatestLogEventSigsAddrsWithConfs(f.latestBlock, AuthTopics, addrs, 0)
+			if err != nil {
+				f.logger.Warnf("Failed to retrieve latest log round %s", err)
+				continue
+			}
+			if len(logs) == 0 {
+				f.logger.Debugf("Empty auth update round for addrs: %s, skipping", addrs)
+				continue
+			}
+
+			f.logger.Infof("handling new %d auth updates", len(logs))
+
+			for _, log := range logs {
+				if err = f.handleAuthChange(log); err != nil {
+					f.logger.Warnf("Error handling auth change %s: %s", log.TxHash, err)
+				}
+
+			}
+
+		case <-f.chStop:
+			return
+		}
+	}
+}
+
+func (f *FwdMgr) handleAuthChange(log evmlogpoller.Log) error {
+	if f.latestBlock > log.BlockNumber {
+		return nil
+	}
+
+	f.latestBlock = log.BlockNumber
+
+	ethLog := types.Log{
+		Address:   log.Address,
+		Data:      log.Data,
+		Topics:    topics(log),
+		TxHash:    log.TxHash,
+		BlockHash: log.BlockHash,
+	}
+
+	switch {
+	case ethLog.Topics[0] == AuthTopics[0]:
+		event, err := f.authRcvr.ParseAuthorizedSendersChanged(ethLog)
+		if err != nil {
+			return errors.Errorf("Failed to parse senders change log")
+		}
+		f.setCachedSenders(event.Raw.Address, event.Senders)
+	case ethLog.Topics[0] == AuthTopics[1]:
+		// ConfigSet event
+		event, err := f.offchainAgg.ParseConfigSet(ethLog)
+		if err != nil {
+			return errors.Errorf("Failed to parse config set log")
+		}
+		f.setCachedSenders(event.Raw.Address, event.Transmitters)
+	}
+
+	return nil
+}
+
+func topics(l evmlogpoller.Log) []common.Hash {
+	var tps []common.Hash
+	for _, topic := range l.Topics {
+		tps = append(tps, common.BytesToHash(topic))
+	}
+	return tps
+}
+
+func (f *FwdMgr) collectAddresses() (addrs []common.Address) {
+	f.cacheMu.RLock()
+	defer f.cacheMu.RUnlock()
+	for addr := range f.sendersCache {
+		addrs = append(addrs, addr)
+	}
+	return
+}
+
+// Stop cancels all outgoings calls and stops internal ticker loop.
+func (f *FwdMgr) Stop() {
```

Comment:
Could embed the standard `utils.StartStopOnce` for extra safety in here and Start().

Reply:
Done